### DISCUSSION TITLE
Why isn't mutable default value (`field = List[int] = []`) a documented feature?

### DISCUSSION THREAD

**Discussion by nonZero** [pydantic/pydantic#3877] [Question] [answered]
Hi!

I was surprised a `List[...]` can get an empty list as a default value correctly - every instance will gets its own copy (dataclasses requires to use a `default_factory`).
```
class ShoppingCart(pydantic.BaseModel):
    items: List[Item] = []
```
I didn't find any documentation for this behavior.

I found this question on Stack Overflow:
https://stackoverflow.com/questions/63793662/how-to-give-a-pydantic-list-field-a-default-value

Shouldn't this be documented?

Thanks!

**Comment by teners** [comment, marked as answer]
seems that it's been documented in #6122

upd: [Mutable default values](https://docs.pydantic.dev/latest/concepts/fields/#mutable-default-values) in the docs

**Reply by sydney-runkle** [reply]
Indeed - the changes associated with that PR can be found [here](https://docs.pydantic.dev/latest/usage/models/#fields-with-non-hashable-default-values) on the usage page of the docs.

**Reply by mmohaveri** [reply]
For anyone who may be looking for the aforementioned document, it's been moved to [here](https://github.com/pydantic/pydantic/blob/main/docs/concepts/fields.md#mutable-default-values).
